Use Janssson API instead of YAJL

lsm5 opened the pull request:
YAJL upstream has been pretty much dead for a while now.
I used commit bd1c5c9 in libguestfs source for reference.

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

modified:   Makefile.am
modified:   configure.ac
modified:   src/oci-umount.c

@rhvgoyal @rhatdan docker [run,exec,commit,push] work fine with this. PTAL.

rhatdan commented on src/oci-umount.c:
```diff
@@ -905,7 +896,7 @@ char *getJSONstring(FILE *from, size_t chunksize, char *msg)
 	struct stat stat_buf;
 	char *err = NULL, *JSONstring = NULL;
-	size_t nbytes, bufsize;
+    size_t nbytes, bufsize;
```
Fix tabs to match current behaviour

lsm5 replied:
ack, will push new commit with tabbing fixed. Thanks

rhatdan commented:
We will need to test this manually, but once the indenting is fixed this LGTM
@rhvgoyal PTAL

rhvgoyal commented on src/oci-umount.c:
```diff
@@ -986,27 +977,27 @@ char *getJSONstring(FILE *from, size_t chunksize, char *msg)
 	_cleanup_fclose_ FILE *fp = NULL;
 
 	/* 'bundle' must be specified for the OCI hooks, and from there we read the configuration file */
 	const char *bundle_path[] = { "bundle", (const char *)0 };
```
Are we using bundle_path anymore?

rhvgoyal commented on src/oci-umount.c:
```diff
@@ -986,27 +977,27 @@ char *getJSONstring(FILE *from, size_t chunksize, char *msg)
-	yajl_val v_bundle_path = yajl_tree_get(node, bundle_path, yajl_t_string);
+    json_t *v_bundle_path = json_object_get(node, "bundle");
 	if (!v_bundle_path) {
 		const char *bundle_path[] = { "bundlePath", (const char *)0 };
```
same here. Is this bundle_path variable being used?

rhvgoyal commented on src/oci-umount.c:
```diff
@@ -1019,17 +1010,17 @@ static int parseBundle(const char *id, yajl_val *node_ptr, char **rootfs, struct
-	snprintf(errbuf, BUFLEN, "failed to read config data from %s", config_file_name);
-	configData = getJSONstring(fp, (size_t)CHUNKSIZE, errbuf);
+    snprintf(err.text, sizeof(err.text), "failed to read config data from %s", config_file_name);
+    configData = getJSONstring(fp, (size_t)CHUNKSIZE, err.text);
```
All this is part of generic cleanup and has nothing to do with json conversion? Can we break the patch down in to a patch series and all the changes which are cleanup and not related to json converstion, move them into a separate patch. That way reviewing becomes easier.

lsm5 commented:
@rhatdan I pushed a new branch to address indentation issues.

@rhvgoyal TBH, my approach so far was switching yajl with jansson api without making any additional changes, and getting docker to work with the new executable, but let me dig into this further and try to address your comments. Thanks

rhvgoyal commented on src/oci-umount.c:
```diff
@@ -1065,13 +1056,16 @@ static int parseBundle(const char *id, yajl_val *node_ptr, char **rootfs, struct
 
 	/* Extract values from the config json */
 	const char *mount_points_path[] = {"mounts", (const char *)0 };
```
mount_point_paths not being used anymore?

rhvgoyal commented:
@lsm5, once you push your new changes I will review again.

lsm5 committed:
Use Janssson API instead of YAJL

YAJL upstream has been pretty much dead for a while now.
I used commit bd1c5c9 in libguestfs source for reference.

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

modified:   Makefile.am
modified:   configure.ac
modified:   src/oci-umount.c
